src/instructlab/model/list.py

```diff
@@ -0,0 +1,138 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+import logging
+import os
+import time
+
+# Third Party
+import click
+import pandas
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+from instructlab.model.backends.backends import is_model_gguf, is_model_safetensors
+
+logger = logging.getLogger(__name__)
+
+
+@click.command(name="list")
+@clickext.display_params
+@click.option(
+    "--model-dirs",
+    help="Base directories where models are stored.",
+    multiple=True,
+    default=[DEFAULTS.MODELS_DIR],
+    show_default=True,
+)
+@click.option(
+    "--list-checkpoints",
+    is_flag=True,
+)
+def model_list(model_dirs: list[str], list_checkpoints: bool):
+    """lists models"""
+    data = []
+    for directory in model_dirs:
+        for entry in Path(directory).iterdir():
+            # if file, just tally the size. This must be a GGUF.
+            if entry.is_file() and is_model_gguf(entry):
+                model, age, size = _analyze_gguf(entry)
+                data.append([model, age, size])
+            elif entry.is_dir():
+                for m in _analyze_dir(entry, list_checkpoints, directory):
+                    data.append(m)
+    print_table(["Model Name", "Model Age", "Model Size"], data)
+
+
+# convert_bytes_to_proper_mag takes a dir/file size in Bytes and converts it to the proper Magnitude (MB, GB)
+def convert_bytes_to_proper_mag(f_size: float):
+    magnitudes = ["KB", "MB", "GB"]
```

Comment:
```suggestion
    magnitudes = ["KiB", "MiB", "GiB", "TiB"]
```

Since you are using 1024 you are not printing Gygabyte but Gigibyte and so on. Let's add TiB too.